### PR TITLE
name_or_id clean up for MultiTag

### PR DESCRIPTION
**Pull request by asobolev**
TODO: determine whether removeReference should also accept name

### CODE REVIEW

File: test/TestMultiTag.cpp [hunk 1]
```diff
@@ -171,6 +171,11 @@ void TestMultiTag::testReferences(){
     dt.addReference(da_2);
     CPPUNIT_ASSERT_MESSAGE(counterrmsg.str(), dt.referenceCount() == 2);
 
+    std::stringstream haserrmsg;
+    haserrmsg << "TestMultiTag::testReference: Has method did not work!";
```
Comment (gicmo) [inline test/TestMultiTag.cpp]: Couldn't that just be `std::string haserrmsg = "TestMultiTag::testReference: Has method did not work!"; `

Comment (asobolev) [inline test/TestMultiTag.cpp]: Coulb well be. But there is this stream everywhere, so I don't want to mess it up

Comment (jgrewe) [inline test/TestMultiTag.cpp]: That was the old style before we switched to use the CPPUNIT_ASSERT without message. 
Actually we could simply ```` CPPUNIT_ASSERT_MESSAGE("some message", x == y)```` if we want to keep the messages. 
imho, we can live with that style.  I kind of like the messages but probably they are more work than they are worth.

Comment (asobolev) [inline test/TestMultiTag.cpp]: I'd maybe keep it consistent - having stream for now and then change it to CPPUNIT_ASSERT_MESSAGE("some message", x == y) everywhere, if it is worth so. But then maybe in the separate PR.

**Pull request merged**